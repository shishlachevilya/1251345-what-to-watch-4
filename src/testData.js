export const movies = [
  {
    id: `1`,
    bg: `bg`,
    title: `Mindhunter`,
    videoPrev: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    preview: `img/mindhunter.jpg`,
    poster: `img/mindhunter.jpg`,
    background: `img/mindhunter.jpg`,
    genre: `Thrillers`,
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent varius quis lorem id iaculis. Sed eget luctus orci. Donec cursus lectus auctor quam tempus ultricies. Nullam at aliquam massa. Donec mollis quis urna eget condimentum. Mauris at dictum ante. Phasellus maximus massa augue, ut luctus dolor rutrum et. Aliquam faucibus turpis lectus, sed hendrerit dolor malesuada at.`,
    rating: 10.0,
    votes: 32,
    director: `Best Director`,
    starring: [`Nicola Cage`, `Cara Delvin`, `Joseph Moser`],
    duration: 48,
    release: 2016,
    reviews: [
      {
        text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent varius quis lorem id iaculis. Sed eget luctus orci.`,
        author: `Camille Guaty`,
        rating: 8.4,
        date: `2019-05-08`,
      },
      {
        text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent varius quis lorem id iaculis. Sed eget luctus orci.`,
        author: `John Dou`,
        rating: 3.4,
        date: `2019-05-08`,
      },
      {
        text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent varius quis lorem id iaculis. Sed eget luctus orci.`,
        author: `Chris Hemsworth`,
        rating: 5.2,
        date: `2019-05-08`,
      },
      {
        text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent varius quis lorem id iaculis. Sed eget luctus orci.`,
        author: `Camille Guaty`,
        rating: 8.4,
        date: `2019-05-08`,
      },
      {
        text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent varius quis lorem id iaculis. Sed eget luctus orci.`,
        author: `John Dou`,
        rating: 3.4,
        date: `2019-05-08`,
      },
      {
        text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent varius quis lorem id iaculis. Sed eget luctus orci.`,
        author: `Chris Hemsworth`,
        rating: 5.2,
        date: `2019-05-08`,
      }
    ]
  }
];

export const TabLabel = {
  OVERVIEW: `overview`,
  DETAILS: `details`,
  REVIEWS: `reviews`
};

export const genreList = [
  `All genres`,
  `Comedies`,
  `Crime`,
  `Documentary`,
  `Dramas`,
  `Horror`,
  `Kids & Family`,
  `Romance`,
  `Sci-Fi`,
  `Thrillers`
];

export default {
  movies,
  TabLabel,
  genreList
};
