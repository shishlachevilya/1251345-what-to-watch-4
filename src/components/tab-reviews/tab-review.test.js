import React from "react";
import renderer from "react-test-renderer";
import TabReviews from "./tab-reviews";
import {movies} from "../../testData";

const movie = movies[0];

describe(`TabReviews component test`, () => {
  it(`component should renders correctly`, () => {
    const tree = renderer
    .create(
        <TabReviews movie={movie}/>
    )
    .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
