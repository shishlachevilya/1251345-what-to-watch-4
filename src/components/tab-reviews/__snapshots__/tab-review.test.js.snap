// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabReviews component test component should renders correctly 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent varius quis lorem id iaculis. Sed eget luctus orci.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Camille Guaty
          </cite>
          <time
            className="review__date"
            dateTime="2019-05-08"
          >
            2019-05-08
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        8.4
      </div>
    </div>
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent varius quis lorem id iaculis. Sed eget luctus orci.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Chris Hemsworth
          </cite>
          <time
            className="review__date"
            dateTime="2019-05-08"
          >
            2019-05-08
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        5.2
      </div>
    </div>
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent varius quis lorem id iaculis. Sed eget luctus orci.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            John Dou
          </cite>
          <time
            className="review__date"
            dateTime="2019-05-08"
          >
            2019-05-08
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        3.4
      </div>
    </div>
  </div>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent varius quis lorem id iaculis. Sed eget luctus orci.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            John Dou
          </cite>
          <time
            className="review__date"
            dateTime="2019-05-08"
          >
            2019-05-08
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        3.4
      </div>
    </div>
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent varius quis lorem id iaculis. Sed eget luctus orci.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Camille Guaty
          </cite>
          <time
            className="review__date"
            dateTime="2019-05-08"
          >
            2019-05-08
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        8.4
      </div>
    </div>
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent varius quis lorem id iaculis. Sed eget luctus orci.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Chris Hemsworth
          </cite>
          <time
            className="review__date"
            dateTime="2019-05-08"
          >
            2019-05-08
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        5.2
      </div>
    </div>
  </div>
</div>
`;
